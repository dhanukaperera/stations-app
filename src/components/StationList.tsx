import { Station } from "../types/APIResponseTypes";
import StationItem from "./StationItem";

type StationListProps = {
	stationList: Station[]
}

const StationList = (props: StationListProps) => {

	const { stationList } = props

	if (stationList.length === 0) {
		return <h1>No Stations Found!</h1>
	}

	return <>
		{stationList.map((station) => {
			return <StationItem key={station.id} name={station.name} tagline={station.tagline} brandSlug={station.brand.slug} slug={station.slug} />
		})}
	</>
}

export default StationList