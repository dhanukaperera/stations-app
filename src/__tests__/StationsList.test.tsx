import { test, expect } from 'vitest';
import { render } from '@testing-library/react';
import StationList from '../components/StationList';
import { StaticRouter } from 'react-router-dom/server';


test("it should render a empty stations list", async () => {
	const { asFragment } = render(<StationList stationList={[]} />);
	expect(asFragment()).toMatchSnapshot();
})

test("it should render a list of", () => {

	const mockStationList = [
		{
			brand: {
				id: "KLo",
				slug: "capital"
			},
			gduid: "363e5e3b-1c83-4fc5-87b5-4b4ca4543a1f",
			heraldId: 25,
			id: "AMpR",
			name: "Capital Teesside",
			slug: "teesside",
			streamUrl: "https://media-ssl.musicradio.com/CapitalTeesside",
			tagline: "The UK's No.1 Hit Music Station"
		},
		{
			brand: {
				id: "KLo",
				slug: "capital"
			},
			gduid: "82ca7a98-e7ab-4bf2-b51b-f615a794b453",
			heraldId: 26,
			id: "AMpS",
			name: "Capital Tyne and Wear",
			slug: "tynewear",
			streamUrl: "https://media-ssl.musicradio.com/CapitalTyneWear",
			tagline: "The UK's No.1 Hit Music Station"
		},

	]

	const { asFragment } = render(
		<StaticRouter location={"/"} >
			<StationList stationList={mockStationList} />
		</StaticRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});