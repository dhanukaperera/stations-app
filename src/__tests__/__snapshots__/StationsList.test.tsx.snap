// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render empty Stations List 1`] = `
<DocumentFragment>
  <h1>
    No Stations Found!
  </h1>
</DocumentFragment>
`;

exports[`is should render a emp 1`] = `
<DocumentFragment>
  <h1>
    No Stations Found!
  </h1>
</DocumentFragment>
`;

exports[`is sould  1`] = `
<DocumentFragment>
  <h1>
    No Stations Found!
  </h1>
</DocumentFragment>
`;

exports[`it should render a empty stations list 1`] = `
<DocumentFragment>
  <h1>
    No Stations Found!
  </h1>
</DocumentFragment>
`;

exports[`it should render a list of 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB oSqvK"
  >
    <h2
      data-testid="station-item-name"
    >
       
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="2"
        />
        <path
          d="M16.24 7.76a6 6 0 0 1 0 8.49m-8.48-.01a6 6 0 0 1 0-8.49m11.31-2.82a10 10 0 0 1 0 14.14m-14.14 0a10 10 0 0 1 0-14.14"
        />
      </svg>
       Capital Teesside 
    </h2>
    <p
      data-testid="station-item-tagline"
    >
      The UK's No.1 Hit Music Station
    </p>
    <a
      href="/station/capital/teesside"
    >
      <button
        class="sc-gueYoa jQMzCE"
      >
        More 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M284.9 412.6l138.1-134c6-5.8 9-13.7 9-22.4v-.4c0-8.7-3-16.6-9-22.4l-138.1-134c-12-12.5-31.3-12.5-43.2 0-11.9 12.5-11.9 32.7 0 45.2l83 79.4h-214c-17 0-30.7 14.3-30.7 32 0 18 13.7 32 30.6 32h214l-83 79.4c-11.9 12.5-11.9 32.7 0 45.2 12 12.5 31.3 12.5 43.3 0z"
          />
        </svg>
         
      </button>
    </a>
  </div>
  <div
    class="sc-beqWaB oSqvK"
  >
    <h2
      data-testid="station-item-name"
    >
       
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="2"
        />
        <path
          d="M16.24 7.76a6 6 0 0 1 0 8.49m-8.48-.01a6 6 0 0 1 0-8.49m11.31-2.82a10 10 0 0 1 0 14.14m-14.14 0a10 10 0 0 1 0-14.14"
        />
      </svg>
       Capital Tyne and Wear 
    </h2>
    <p
      data-testid="station-item-tagline"
    >
      The UK's No.1 Hit Music Station
    </p>
    <a
      href="/station/capital/tynewear"
    >
      <button
        class="sc-gueYoa jQMzCE"
      >
        More 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M284.9 412.6l138.1-134c6-5.8 9-13.7 9-22.4v-.4c0-8.7-3-16.6-9-22.4l-138.1-134c-12-12.5-31.3-12.5-43.2 0-11.9 12.5-11.9 32.7 0 45.2l83 79.4h-214c-17 0-30.7 14.3-30.7 32 0 18 13.7 32 30.6 32h214l-83 79.4c-11.9 12.5-11.9 32.7 0 45.2 12 12.5 31.3 12.5 43.3 0z"
          />
        </svg>
         
      </button>
    </a>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with some  1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB oSqvK"
  >
    <h2
      data-testid="station-item-name"
    >
       
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="2"
        />
        <path
          d="M16.24 7.76a6 6 0 0 1 0 8.49m-8.48-.01a6 6 0 0 1 0-8.49m11.31-2.82a10 10 0 0 1 0 14.14m-14.14 0a10 10 0 0 1 0-14.14"
        />
      </svg>
       Capital Teesside 
    </h2>
    <p
      data-testid="station-item-tagline"
    >
      The UK's No.1 Hit Music Station
    </p>
    <a
      href="/station/capital/teesside"
    >
      <button
        class="sc-gueYoa jQMzCE"
      >
        More 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M284.9 412.6l138.1-134c6-5.8 9-13.7 9-22.4v-.4c0-8.7-3-16.6-9-22.4l-138.1-134c-12-12.5-31.3-12.5-43.2 0-11.9 12.5-11.9 32.7 0 45.2l83 79.4h-214c-17 0-30.7 14.3-30.7 32 0 18 13.7 32 30.6 32h214l-83 79.4c-11.9 12.5-11.9 32.7 0 45.2 12 12.5 31.3 12.5 43.3 0z"
          />
        </svg>
         
      </button>
    </a>
  </div>
  <div
    class="sc-beqWaB oSqvK"
  >
    <h2
      data-testid="station-item-name"
    >
       
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="2"
        />
        <path
          d="M16.24 7.76a6 6 0 0 1 0 8.49m-8.48-.01a6 6 0 0 1 0-8.49m11.31-2.82a10 10 0 0 1 0 14.14m-14.14 0a10 10 0 0 1 0-14.14"
        />
      </svg>
       Capital Tyne and Wear 
    </h2>
    <p
      data-testid="station-item-tagline"
    >
      The UK's No.1 Hit Music Station
    </p>
    <a
      href="/station/capital/tynewear"
    >
      <button
        class="sc-gueYoa jQMzCE"
      >
        More 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M284.9 412.6l138.1-134c6-5.8 9-13.7 9-22.4v-.4c0-8.7-3-16.6-9-22.4l-138.1-134c-12-12.5-31.3-12.5-43.2 0-11.9 12.5-11.9 32.7 0 45.2l83 79.4h-214c-17 0-30.7 14.3-30.7 32 0 18 13.7 32 30.6 32h214l-83 79.4c-11.9 12.5-11.9 32.7 0 45.2 12 12.5 31.3 12.5 43.3 0z"
          />
        </svg>
         
      </button>
    </a>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with some pets 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB oSqvK"
  >
    <h2
      data-testid="station-item-name"
    >
       
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="2"
        />
        <path
          d="M16.24 7.76a6 6 0 0 1 0 8.49m-8.48-.01a6 6 0 0 1 0-8.49m11.31-2.82a10 10 0 0 1 0 14.14m-14.14 0a10 10 0 0 1 0-14.14"
        />
      </svg>
       Capital Teesside 
    </h2>
    <p
      data-testid="station-item-tagline"
    >
      The UK's No.1 Hit Music Station
    </p>
    <a
      href="/station/capital/teesside"
    >
      <button
        class="sc-gueYoa jQMzCE"
      >
        More 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M284.9 412.6l138.1-134c6-5.8 9-13.7 9-22.4v-.4c0-8.7-3-16.6-9-22.4l-138.1-134c-12-12.5-31.3-12.5-43.2 0-11.9 12.5-11.9 32.7 0 45.2l83 79.4h-214c-17 0-30.7 14.3-30.7 32 0 18 13.7 32 30.6 32h214l-83 79.4c-11.9 12.5-11.9 32.7 0 45.2 12 12.5 31.3 12.5 43.3 0z"
          />
        </svg>
         
      </button>
    </a>
  </div>
  <div
    class="sc-beqWaB oSqvK"
  >
    <h2
      data-testid="station-item-name"
    >
       
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="2"
        />
        <path
          d="M16.24 7.76a6 6 0 0 1 0 8.49m-8.48-.01a6 6 0 0 1 0-8.49m11.31-2.82a10 10 0 0 1 0 14.14m-14.14 0a10 10 0 0 1 0-14.14"
        />
      </svg>
       Capital Tyne and Wear 
    </h2>
    <p
      data-testid="station-item-tagline"
    >
      The UK's No.1 Hit Music Station
    </p>
    <a
      href="/station/capital/tynewear"
    >
      <button
        class="sc-gueYoa jQMzCE"
      >
        More 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M284.9 412.6l138.1-134c6-5.8 9-13.7 9-22.4v-.4c0-8.7-3-16.6-9-22.4l-138.1-134c-12-12.5-31.3-12.5-43.2 0-11.9 12.5-11.9 32.7 0 45.2l83 79.4h-214c-17 0-30.7 14.3-30.7 32 0 18 13.7 32 30.6 32h214l-83 79.4c-11.9 12.5-11.9 32.7 0 45.2 12 12.5 31.3 12.5 43.3 0z"
          />
        </svg>
         
      </button>
    </a>
  </div>
</DocumentFragment>
`;

exports[`renders correctly with some stations 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB oSqvK"
  >
    <h2
      data-testid="station-item-name"
    >
       
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="2"
        />
        <path
          d="M16.24 7.76a6 6 0 0 1 0 8.49m-8.48-.01a6 6 0 0 1 0-8.49m11.31-2.82a10 10 0 0 1 0 14.14m-14.14 0a10 10 0 0 1 0-14.14"
        />
      </svg>
       Capital Teesside 
    </h2>
    <p
      data-testid="station-item-tagline"
    >
      The UK's No.1 Hit Music Station
    </p>
    <a
      href="/station/capital/teesside"
    >
      <button
        class="sc-gueYoa jQMzCE"
      >
        More 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M284.9 412.6l138.1-134c6-5.8 9-13.7 9-22.4v-.4c0-8.7-3-16.6-9-22.4l-138.1-134c-12-12.5-31.3-12.5-43.2 0-11.9 12.5-11.9 32.7 0 45.2l83 79.4h-214c-17 0-30.7 14.3-30.7 32 0 18 13.7 32 30.6 32h214l-83 79.4c-11.9 12.5-11.9 32.7 0 45.2 12 12.5 31.3 12.5 43.3 0z"
          />
        </svg>
         
      </button>
    </a>
  </div>
  <div
    class="sc-beqWaB oSqvK"
  >
    <h2
      data-testid="station-item-name"
    >
       
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="2"
        />
        <path
          d="M16.24 7.76a6 6 0 0 1 0 8.49m-8.48-.01a6 6 0 0 1 0-8.49m11.31-2.82a10 10 0 0 1 0 14.14m-14.14 0a10 10 0 0 1 0-14.14"
        />
      </svg>
       Capital Tyne and Wear 
    </h2>
    <p
      data-testid="station-item-tagline"
    >
      The UK's No.1 Hit Music Station
    </p>
    <a
      href="/station/capital/tynewear"
    >
      <button
        class="sc-gueYoa jQMzCE"
      >
        More 
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M284.9 412.6l138.1-134c6-5.8 9-13.7 9-22.4v-.4c0-8.7-3-16.6-9-22.4l-138.1-134c-12-12.5-31.3-12.5-43.2 0-11.9 12.5-11.9 32.7 0 45.2l83 79.4h-214c-17 0-30.7 14.3-30.7 32 0 18 13.7 32 30.6 32h214l-83 79.4c-11.9 12.5-11.9 32.7 0 45.2 12 12.5 31.3 12.5 43.3 0z"
          />
        </svg>
         
      </button>
    </a>
  </div>
</DocumentFragment>
`;
